src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -1425,10 +1428,50 @@ void SystemLayout::doLayoutNoteSpannersLinear(System* system, LayoutContext& ctx
                 Chord* c = toChord(e);
                 for (Chord* ch : c->graceNotes()) {
                     layoutTies(ch, system, start, ctx);
-                    layoutNoteAnchoredSpanners(system, ch);
                 }
                 layoutTies(c, system, start, ctx);
-                layoutNoteAnchoredSpanners(system, c);
+            }
+        }
+    }
+}
+
+void SystemLayout::setNoteSpannersSystem(System* system, const bool isLinearMode)
+{
+    auto setSystem = [&] (Chord* chord) {
+        // Add all spanners attached to notes, otherwise these will be removed if outside of the layout range
+        for (Note* note : chord->notes()) {
+            for (Spanner* spanner : note->spannerFor()) {
+                for (SpannerSegment* spannerSeg : spanner->spannerSegments()) {
+                    if (!isLinearMode && !spannerSeg->isSingleBeginType()) {
+                        continue;
+                    }
+                    spannerSeg->setSystem(system);
```

Comment:
Actually, now that I look at it more closely... It is a bit weird to be doing this operation here. System is the parent item of SpannerSegment, and normally SpannerSegments get their parent set right at the moment that they are created. If we get here and these spannerSegments don't have their parent correctly set, doesn't it mean that something went wrong earlier? Should this be fixed at an earlier point?

Reply:
Systems are set during layout eg. `TLayout::layoutLine(SLine...` and for note anchored spanners, only the spanners within the layout range are layed out again.  However, every system the layout range spans is totally reset (`System::clear`) which removes all spanner segments.  
The disappearing glisses fall outside the layout range but are on a system which is reset, and thus aren't drawn.
Do you think it would be better to only clear systems within the layout range?  This would remove the need for this method.

Reply:
Ok I think I've figured out the problem. Glissandos have some peculiarities with respect to other spanners. 

First is that we lay them out at the Page level (`PageLayout::collectPage` calls `ChordLayout::layoutSpanners`), and that's because they often cross staves which means they need to be done after vertical justification. 

Second the fact that if they cross a system- or page-break, their two segments aren't laid out independently like most other spanners, because they need to know about each other to match their slant etc. That's why when we call layout on the Glissando we always layout _both_ of its GlissandoSegments. This means that, to avoid calling layout twice, we either call it from its start note (by looking at the SpannerFor of that note) or from its end note (by looking at the SpannerBack of that note). It's indifferent really which one to use, but it's usually safer to call it from the end note because it ensures that the start note has already been laid out and we know its position. 

Here the problem is simply that the last page that gets laid out doesn't contain the end note of the glissando, and as a result the glissando doesn't get laid out and disappears. As far as I could tell from a quick test, making this simple addition to ChordLayout::layoutSpanners which accounts for the case of cross-page glissandos fixes the problem.

![image](https://github.com/user-attachments/assets/834d931f-ff4e-4567-a3e4-60f67ba15894)

Reply:
Great, this sorts the issue out for glisses over page breaks.  The exception for guitar bends above is causing them to disappear on system breaks, so I have also removed that.  
For horizontal view, I can keep the solution in https://github.com/musescore/MuseScore/pull/23418 or to keep things consistent I can call `layoutSpanners` on all chords in the score (in `layoutLinear`)?  This feels like a lot of layout that isn't really necessary however.

Reply:
Horizontal view is a unique mess of its own, due to it being a single large system but _some_ of the layout operations being done only on the region that's currently on screen. So I'd say let's keep the current solution there and touch it as least as possible. We'll come back to that in the future to properly sort it out